From eecd56d89df0ca1f01f847e53e27fe6781585a0d Mon Sep 17 00:00:00 2001
From: David Bastos <david.barbosabastos@windriver.com>
Date: Wed, 29 Apr 2026 09:18:54 -0300
Subject: [PATCH 2/2] Adopt StarlingX image format in values.yaml

Reorganize image references under spo.image and krp.image keys,
following the StarlingX expected format so the platform can correctly
prefix images with the local registry.

Signed-off-by: David Bastos <david.barbosabastos@windriver.com>
---
 templates/deployment.yaml |  4 ++--
 values.yaml               | 11 ++++++++---
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/templates/deployment.yaml b/templates/deployment.yaml
index bf5c841..9ea026c 100644
--- a/templates/deployment.yaml
+++ b/templates/deployment.yaml
@@ -19,7 +19,7 @@ spec:{{if not .Values.autoscaling.enabled}}
         - manager
         env:
         - name: RELATED_IMAGE_RBAC_PROXY
-          value: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
+          value: {{.Values.krp.image.repository}}:{{.Values.krp.image.tag}}
         - name: RELATED_IMAGE_SELINUXD
           value: quay.io/security-profiles-operator/selinuxd
         - name: RELATED_IMAGE_SELINUXD_EL8
@@ -30,7 +30,7 @@ spec:{{if not .Values.autoscaling.enabled}}
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: {{.Values.spoImage.repository}}:{{.Values.spoImage.tag}}
+        image: {{.Values.spo.image.repository}}:{{.Values.spo.image.tag}}
         imagePullPolicy: Always
         name: {{.Chart.Name}}
         resources:{{toYaml .Values.resources | nindent 10}}
diff --git a/values.yaml b/values.yaml
index aa294da..d56ca82 100644
--- a/values.yaml
+++ b/values.yaml
@@ -1,9 +1,14 @@
 # Default values for security-profiles-operator.
 replicaCount: 3
 
-spoImage:
-  repository: docker.io/starlingx/security-profiles-operator
-  tag: stx.12.0-v0.8.3-1
+spo:
+  image:
+    repository: docker.io/starlingx/security-profiles-operator
+    tag: stx.12.0-v0.8.3-1
+krp:
+  image:
+    repository: gcr.io/kubebuilder/kube-rbac-proxy
+    tag: v0.15.0
 
 enableSelinux: false
 enableLogEnricher: false
--
2.43.0
